src/Hazelcast.Net.Tests/Serialization/Compact/GenericRecordTests.cs

```diff
@@ -0,0 +1,410 @@
+        // the compact type "rec0", we fall back to generic record
+        var rec0B = await mapOfGenericRecord.GetAsync(0);
+        Assert.That(rec0B, Is.Not.Null);
+        Assert.That(rec0B, Is.InstanceOf<CompactGenericRecordBase>());
+        Assert.That(rec0B, Is.InstanceOf<CompactDictionaryGenericRecord>());
+        Assert.That(rec0B.GetString("string"), Is.EqualTo("hello"));
+
+        await mapOfObject.DisposeAsync();
+    }
+
+    [Test]
+    public void CoverGenericRecord()
+    {
+        var schema = SchemaBuilder.For("type-name")
+            .WithField("field-name-1", FieldKind.Boolean)
+            .WithField("field-name-2", FieldKind.Int8)
+            .WithField("field-name-3", FieldKind.String)
+            .Build();
+
+        var fieldValues = new Dictionary<string, object?>
+        {
+            {"field-name-1", true},
+            {"field-name-2", 123},
+            {"field-name-3", "hello"}
+        };
+
+        var rec = new CompactDictionaryGenericRecord(schema, fieldValues);
+
+        Assert.That(rec.GetFieldKind("field-name-1"), Is.EqualTo(FieldKind.Boolean));
+        Assert.That(rec.GetFieldKind("field-name-2"), Is.EqualTo(FieldKind.Int8));
+        Assert.That(rec.GetFieldKind("field-name-3"), Is.EqualTo(FieldKind.String));
+
+        var fieldNames = rec.FieldNames;
+        Assert.That(fieldNames.Count, Is.EqualTo(3));
+        Assert.That(fieldNames, Does.Contain("field-name-1"));
+        Assert.That(fieldNames, Does.Contain("field-name-2"));
+        Assert.That(fieldNames, Does.Contain("field-name-3"));
+
+        Assert.Throws<ArgumentNullException>(() => _ = new CompactDictionaryGenericRecord(null!, fieldValues));
+        Assert.Throws<ArgumentNullException>(() => _ = new CompactDictionaryGenericRecord(schema, null!));
+    }
+
+    [Test]
+    public void CoverGenericRecordBuilder()
+    {
+        var schema = SchemaBuilder.For("type-name")
+            .WithField("field-name-1", FieldKind.Boolean)
+            .WithField("field-name-2", FieldKind.Int8)
+            .WithField("field-name-3", FieldKind.String)
+            .Build();
+
+        Assert.Throws<ArgumentException>(() => _ = new CompactGenericRecordBuilder((string) null!));
+        Assert.Throws<ArgumentException>(() => _ = new CompactGenericRecordBuilder(""));
+        Assert.Throws<ArgumentNullException>(() => _ = new CompactGenericRecordBuilder((Schema)null!));
+        Assert.Throws<ArgumentNullException>(() => _ = new CompactGenericRecordBuilder(schema, null!));
+
+        var builder = new CompactGenericRecordBuilder(schema);
+
+        // getting an ArgumentException with "should not be null nor empty" in both cases
+        // we don't differentiate null & empty string exception to keep it simple
+        Assert.Throws<ArgumentException>(() => builder.SetBoolean(null!, true));
+        Assert.Throws<ArgumentException>(() => builder.SetBoolean("", true));
```

Comment:
Er... WHAT? So in Java it's legal in compact serialization to have a field named `""`? Nowhere in .NET do we support it, we have tests all over the place to ensure that the field name is not empty. Of course we can revert it but it would mean... changing code in many places + I quite don't see the reasoning behind it?

Reply:
This is discussed in slack. Resolving.